apiserver/facades/client/controller/controller.go

```diff
@@ -249,36 +253,21 @@ func (c *ControllerAPI) AllModels(ctx context.Context) (params.UserModelList, er
 		return result, errors.Trace(err)
 	}
 
-	modelUUIDs, err := c.state.AllModelUUIDs()
+	models, err := c.modelService.ListAllModels(ctx)
 	if err != nil {
 		return result, errors.Trace(err)
 	}
-	for _, modelUUID := range modelUUIDs {
-		st, err := c.statePool.Get(modelUUID)
-		if err != nil {
-			// This model could have been removed.
-			if errors.Is(err, errors.NotFound) {
-				continue
-			}
-			return result, errors.Trace(err)
-		}
-		defer st.Release()
-
-		model, err := st.Model()
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-
+	for _, model := range models {
 		userModel := params.UserModel{
 			Model: params.Model{
-				Name:     model.Name(),
-				UUID:     model.UUID(),
-				Type:     string(model.Type()),
-				OwnerTag: model.Owner().String(),
+				Name:     model.Name,
+				UUID:     model.UUID.String(),
+				Type:     model.ModelType.String(),
+				OwnerTag: model.OwnerName.Name(),
```

Comment:
You need to make this a user tag I think. This will be `OwnerTag: names.NewUserTag(model.OwnerName.Name()),`

Reply:
Hi tom, not sure if im understanding correcting but the OwnerTag here is a json string, do you mean 

names.NewUserTag(model.OwnerName.Name()).String()?

Reply:
no you are correct we don't want the panic path. We should use the Parse variant and handle the error.

Reply:
will change this accordingly thanks tom! :D

Reply:
Sure did a IsValidUser check before this and do skip + error log for invalid models, thanks tom!